### DISCUSSION TITLE
Hello and a quick question about custom claims

### DISCUSSION THREAD

**Discussion by ergec** [ezralazuardy/heimdall#10] [Q&A] [answered]
Hi,
First of all thanks for this great library. I always wanted to try and setup my own server, at least testing for now. Heimdall made it so much easier for me to make it up and running. Again thanks a lot. Great name BTW Heimdall.

A quick question. Is it possible to add custom claims to Access Token? Currently my access tokens look like this and it's not very much info .

        {
          "aud": "test",
          "jti": "c501e3867961f7a4a9ff4b592a048deaed17a9dd20e2c3c51102ebde6363151009c4228a229796da",
          "iat": 1616494623,
          "nbf": 1616494623,
          "exp": 1616498223,
          "sub": "1",
          "scopes": [
            "basic",
            "openid"
          ]
        }

Again thanks a lot for Heimdall

**Comment by ezralazuardy** [comment, marked as answer]
sorry for the late response, i've been busy lately.

if the "custom claim" that you mean is a openid related claims, yes Heimdall support it. You can define the prefered claim in `getClaims()` in the UserEntity.

reference: https://heimdall.ezralazuardy.com/documentation/oidc#implement-the-claimsetinterface
